x/concentrated-liquidity/lp.go

```diff
@@ -381,3 +394,39 @@ func emitLiquidityChangeEvent(ctx sdk.Context, eventType string, positionId uint
 		sdk.NewAttribute(types.AttributeAmount1, actualAmount1.String()),
 	))
 }
+
+// validateIsNotLockedAndUpdate checks if the concentrated position's underlying lock has expired
+//   - If it has not expired, then we cannot withdraw from this position
+//   - If it has expired, we update the underlying lock ID to zero to indicate that this position no longer has an underlying lock and we can withdraw from this position
+func (k Keeper) validateIsNotLockedAndUpdate(ctx sdk.Context, position model.Position, underlyingLockId uint64) (model.Position, error) {
```

Comment:
Was actually able to remove this for a single boolean check since we no longer hold the lockID connection in the position struct itself https://github.com/osmosis-labs/osmosis/pull/4854/commits/1c7e1220efa8230c0b86c8040925794262f09336